launchpad/src/launchpad.rs

```diff
@@ -50,22 +50,65 @@ pub trait Launchpad: setup::SetupModule + ongoing_operation::OngoingOperationMod
         Ok(())
     }
 
+    #[only_owner]
+    #[endpoint(addAddressToBlacklist)]
+    fn add_address_to_blacklist(&self, address: Address) -> SCResult<()> {
+        self.blacklist().insert(address);
+
+        Ok(())
+    }
+
+    #[only_owner]
+    #[endpoint(removeAddressFromBlacklist)]
+    fn remove_address_from_blacklist(&self, address: Address) -> SCResult<()> {
+        self.blacklist().remove(&address);
+
+        Ok(())
+    }
+
+    #[only_owner]
+    #[endpoint(refundConfirmedTickets)]
+    fn refund_confirmed_tickets(&self, address: Address) -> SCResult<()> {
+        require!(
+            self.blacklist().contains(&address),
+            "Can only refund for users that have been put in blacklist"
+        );
+
+        let (first_ticket_id, last_ticket_id) = self.ticket_range_for_address(&address).get();
+        let mut nr_refunded_tickets = 0;
+
+        for ticket_id in first_ticket_id..=last_ticket_id {
+            let ticket_status = self.ticket_status(ticket_id).get();
+            if !ticket_status.is_confirmed() {
+                continue;
+            }
+
+            self.ticket_status(ticket_id).set(&TicketStatus::None);
+
+            nr_refunded_tickets += 1;
+        }
+
+        self.total_confirmed_tickets()
+            .update(|confirmed| *confirmed -= nr_refunded_tickets);
+
+        let ticket_paymemt_token = self.ticket_payment_token().get();
+        let ticket_price = self.ticket_price().get();
+        let amount_to_refund = ticket_price * nr_refunded_tickets.into();
+
+        self.send()
+            .direct(&address, &ticket_paymemt_token, 0, &amount_to_refund, &[]);
+
+        Ok(())
+    }
+
     #[only_owner]
     #[endpoint(addTickets)]
     fn add_tickets(
         &self,
         #[var_args] address_number_pairs: VarArgs<MultiArg2<Address, usize>>,
     ) -> SCResult<()> {
-        self.require_no_ongoing_operation()?;
         self.require_stage(LaunchStage::AddTickets)?;
 
```

Comment:
please put back the requires and checks. after the selection process starts, we should not add any more tickets.